Source/WebKit/Platform/cocoa/CocoaHelpers.mm

```diff
@@ -381,6 +381,24 @@ id parseJSON(API::Data& json, JSONOptionSet options, NSError **error)
     return result;
 }
 
+static bool stringHasCaseInsensitivePrefix(NSString *string, NSString *prefix)
+{
+    if (!prefix)
+        return NO;
+
+    return [string rangeOfString:prefix options:NSAnchoredSearch | NSCaseInsensitiveSearch].location != NSNotFound;
+}
+
+bool isHTTPFamilyURL(NSURL *url)
+{
+    NSString *urlString = url.absoluteString;
+    if (!urlString)
+        return NO;
+
+    return stringHasCaseInsensitivePrefix(urlString, @"http:") || stringHasCaseInsensitivePrefix(urlString, @"https:");
+
+}
```

Comment:
We can use WTF URL methods for this check, it can convert from NSURL easily with a cast.

Reply:
I got rid of this helper method and used:

`if (!URL(url).protocolIsInHTTPFamily()) {`

at the callsite.